nixos/modules/services/networking/nomad.nix

```diff
@@ -40,7 +40,7 @@ in
 
       enableDocker = mkOption {
```

Comment:
I'm not even sure that having this open is that useful. It seems to give the illusion that a user only need set this to `true` to allow nomad to launch containers, and that isn't the case. You have to enable to docker task driver plugin.

I think we either remove this option altogether, or we have to add a `client.enabled = true` option (and then probably a `server.enabled = true` option) and then we have to enable the plugin.

Unfortunately, that still probably won't be enough. Users will want to be able to launch containers using images from a private registry, in which case we'd basically have to replicate the nomad client config structure here. Of course, when people see that option they might start expecting that other config options can be set in the same way and before long we've replicated the entire nomad client and server configuration structs here!

I propose that we leave this option out to avoid confusion.

Reply:
> It seems to give the illusion that a user only need set this to `true` to allow nomad to launch containers, and that isn't the case. You have to enable to docker task driver plugin.

@cpcloud For me it worked out of the box, I didn't have to enable a task driver plugin (as stated in https://github.com/NixOS/nixpkgs/pull/105739#issuecomment-759186552 with the config shown in there).

Reply:
Ah interesting! Ok then, I guess I'm wrong here. I agree then that we should keep this option.

Reply:
@lovesegfault @aanderse @cpcloud Given this, what's your opinion on the default?

Mine is that we should default to `true` because docker is the most common way to use Nomad, and one can only play through the official tutorial when it's enabled.

Reply:
@nh2 I agree with you. `enableDocker = true` seems like the right default.